### ISSUE TITLE
kmplot

### ISSUE THREAD

**Issue by JianGuoZhou3** [nanxstats/hdnom#9]
hello @    @@@nan,
I have a new question.
kmplot(cal.int_lasso, group.name = c('High risk', 'Low risk'),time.at = 1:6 * 365)
cal.int.logrank = logrank_test(cal.int_lasso)
cal.int.logrank
cal.int.logrank$pval
#Call:
survdiff(formula = formula("Surv(time, event) ~ grp"))

n=510, 1 observation deleted due to missingness.

        N Observed Expected (O-E)^2/E (O-E)^2/V
grp=1 255      133     78.9      37.1      68.4
grp=2 255       41     95.1      30.8      68.4

 Chisq= 68.4  on 1 degrees of freedom, p= <2e-16 
[1] 1.365953e-16

but i used :nom[["nomogram"]][["coefficients"]]
          Intercept          INSS_Stage              Ploidy Histology_Favorable 
        0.000000000         0.004752014         0.079239332        -0.218644316 
     COG_Risk_Group 
        0.531646760 
so I calculate the result by the following code.
train1=train
train1$risk=train1$INSS_Stage*0.004752014+train1$Ploidy*0.079239332-train1$Histology_Favorable*0.218644316+train1$COG_Risk_Group*0.531646760
median(train1$risk)
train1$risk_group=ifelse(train1$risk>median(train1$risk),"high","low")
table(train1$risk_group)

[1] 1.6187

high  low 
 161  350
but this result did meet your result.
please help me to check this.
best wishes,
jianguo

**Comment by nanxstats** [comment]
Well, I'm not really sure what the question is, but here's how the risk groups are assigned: https://github.com/nanxstats/hdnom/blob/master/R/4_1_calibrate.R

The first step is to cut the samples into a specified number of risk groups based on their predicted survival probability. To do this, you'll need to get the quantiles (cut points) first, then by using `cut()` you can get a group number assigned for each sample. The second one is just using `tapply()` to get the median survival probability within each group.

**Comment by JianGuoZhou3** [comment]
Dear nan,
Thanks for your response.
I think this is a bug.
I just try by the method, in the validation group, your package and my code is consilient.
So, I used my code to calculate  the HR and 95%CI.
And the nomogram just for Overall survival, I also used this nomogram to predict the RFS.
BEST WISHES
Jianguo

 



Jian-Guo ZHOU M.D.
Department of Oncology, Affiliated Hospital of Zunyi Medical University
Address: NO.149, Dalian Road, Zunyi City, Guizhou Province, China, 563000
Email: jianguo.zhou@yahoo.com
